### PR TITLE
[PO-4221] Adding deployment approvers

### PR DESCRIPTION
**Pull request by mbdimitrova**
### What does this PR do?
It adds *repo approvers* setting for repositories ✨ Repo approvers, similar to repo owners, consist of lists with user emails. The purpose of repo approvers is to assure that the ticket is approved in Jira by a user from the list 🛡 This PR only adds the repo approver field in the repository form; the implementation of the functionality will be added in another PR.

### Any background context you want to provide?
🤷‍♀️ 

### Any specific implementation changes that would benefit highlighting?
* `repo_approvers` is added to `repo_ownerships` table
* **Repo approvers** are added to repositories index
* Wraping of text is enabled in **Repo approvers** and **Repo owners** columns of the repositories table to make it more compact

### What are the relevant tickets / PRs?
- [x] JIRA :gem:: https://jira.fundingcircle.com/browse/PO-4221
- [x] *Related*: https://jira.fundingcircle.com/browse/PO-4222

/cc @FundingCircle/prod-ops @0mega @sgerrand 🙇‍♀️

### CODE REVIEW

File: app/models/events/repo_ownership_event.rb [hunk 1]
```diff
@@ -10,5 +10,9 @@ def app_name
     def repo_owners
       details['repo_owners']
     end
+
+    def repo_approvers
```
[the remark this answers is not in the document]
Comment (mbdimitrova) [reply app/models/events/repo_ownership_event.rb]: 👍

**Comment by mbdimitrova** [comment]
@popovm I renamed `RepoOwner` to `RepoAdmin`, I think now it's more understandable 😃

File: app/helpers/repo_admins_helper.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+module RepoAdminsHelper
```
Comment (sgerrand) [inline app/helpers/repo_admins_helper.rb]: 🙋 Why was this renamed? Repository owners are not generally repository administrators. The original name made more sense to me.

File: app/mailers/deploy_alert_mailer.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 class DeployAlertMailer < ApplicationMailer
   # rubocop:disable Metrics/ParameterLists
   def deploy_alert_email(repo_owners:, repo:, region:, deployer:, deployed_at:, alert:, releases_url:)
-    owner_emails = RepoOwner.to_mail_address_list(repo_owners).format
+    owner_emails = RepoAdmin.to_mail_address_list(repo_owners).format
```
Comment (sgerrand) [inline app/mailers/deploy_alert_mailer.rb]: It also prevented weird naming traversals like this.

File: app/models/events/release_exception_event.rb [hunk 4]
```diff
@@ -24,7 +24,7 @@ def git_repos
     end
 
     def repo_owner
-      RepoOwner.find_or_initialize_by(email: email)
+      RepoAdmin.find_or_initialize_by(email: email)
```
Comment (sgerrand) [inline app/models/events/release_exception_event.rb]: ...and this.

File: app/helpers/repo_admins_helper.rb (hunk 2 above)
Comment (popovm) [reply app/helpers/repo_admins_helper.rb]: Initially there was only one "role" required in ST - the repo owner, who is able to justify unapproved deployments. But now we want additional role, who has weaker privileges - to approve tickets in the issue tracker. So we need three terms - one combining both roles, one for the master role(the original repo owners) and one for approvers. 
The idea is that `repo admin` could be the general name, `repo owner` to be the master role and `repo approver` to be the newly introduced role.
Unarguably this renaming was weird and annoying, but leaving repo owners as the general role left us with no name proposals for the current role and would have still required a lot of renaming 😕 
We would be very happy to hear new ideas for organising and naming repo owners and approvers in the project. 🙂
